admission/hooks/validate_pod_test.go

```diff
@@ -90,4 +92,40 @@ spec:
 		permissive := os.Getenv("TEST_PERMISSIVE") == "true"
 		Expect(err == nil).To(Equal(permissive))
 	})
+
+	It("should deny a Pod having untrustworthy Ephemeral Container", func() {
+		podManifest := `apiVersion: v1
+kind: Pod
+metadata:
+  name: test-pod-validate-4
+  namespace: default
+spec:
+  containers:
+  - name: ubuntu
+    image: quay.io/cybozu/ubuntu
+`
+
+		d := yaml.NewYAMLOrJSONDecoder(strings.NewReader(podManifest), 4096)
+		po := &v1.Pod{}
+		err := d.Decode(po)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = k8sClient.Create(testCtx, po)
+		Expect(err).NotTo(HaveOccurred())
+
+		ephemeralContainer := v1.EphemeralContainer{
+			EphemeralContainerCommon: v1.EphemeralContainerCommon{
+				Name:  "debug",
+				Image: "busybox",
+			},
+		}
+		po.Spec.EphemeralContainers = append(po.Spec.EphemeralContainers, ephemeralContainer)
+
+		k8s, err := kubernetes.NewForConfig(k8sConfig)
+		Expect(err).NotTo(HaveOccurred())
+
+		_, err = k8s.CoreV1().Pods(po.Namespace).UpdateEphemeralContainers(testCtx, po.Name, po, metav1.UpdateOptions{})
+		permissive := os.Getenv("TEST_PERMISSIVE") == "true"
+		Expect(err == nil).To(Equal(permissive))
```

Comment:
Is it correct that the meaning of the code here is that it confirms that err is nil?
I ask because I thought that if adding an ephemeral container fails due to admission webhook deny, err would not be nil.

Reply:
When PERMISSIVE mode is enabled, err will not be nil.
However, this test is difficult to understand. I will rewrite it.

Reply:
https://github.com/cybozu/neco-containers/pull/868/commits/0670a214928f09be5adb97d59518c89324763744

Reply:
Thank you for rewriting the code.
It helped me to understand it.